common/version.go

```diff
@@ -96,3 +96,25 @@ func setupUpgradeVersion(cfg *config.Config, osd *osd.OSD) (err error) {
 		cfg.Cluster.Version, cfg.Upgrade.ReleaseName, cfg.Upgrade.ReleaseStream)
 	return
 }
+
+// GetNewestAndOldestVersions returns a list which contains the newest and the oldest non-default versions.
+func GetNewestAndOldestVersions(cfg *config.Config) ([]string, error) {
+	OSD, err := osd.New(cfg.OCM.Token, cfg.OCM.Env, cfg.OCM.Debug)
+	if err != nil {
+		return nil, fmt.Errorf("could not setup OSD: %v", err)
+	}
+
+	versionList, err := OSD.EnabledNoDefaultVersionList()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(versionList) < 2 {
+		return versionList, nil
+	}
+
+	return []string{
+		versionList[0],
+		versionList[len(versionList)-1],
```

Comment:
I wonder if we should pick something in the middle here, e.g. len(versionList)/2) instead of -1. Or something like that -- @jeefy, @yasun1, what do you guys think?

Reply:
I can understand your concern. I think we have covered a kind of the middle one. The middle one is based on where the default version(which is covered by the regular testing) is:
1. The oldest < the default < the newest
2. The default < the oldest < the newest -- the oldest = the default + 1
3. The oldest < the newest < the default  -- the newest =  the default - 1

If the newest in above three scenarios has been covered the other cases, it is better for us to use len(versionList)/2) instead of -1 to cover as many ClusterImageSets as possible.

Reply:
Hey @yasun1. Yeah, so you're right: the other three test scenarios are covered by other test cases, so I think len(versionList)/2 is the way to go. So basically we need the oldest version and the middle version.